cni/nnsgrpcclient/nnsgrpcclient_test.go

```diff
@@ -0,0 +1,117 @@
+package nnsgrpcclient
+
+import (
+	"context"
+	"fmt"
+	"github.com/Azure/azure-container-networking/test/nnsmockserver"
+	"os"
+	"strings"
+	"testing"
+	"time"
+)
+
+var mockserver *nnsmockserver.NnsMockServer
+
+func TestMain(m *testing.M) {
+	setup()
+	code := m.Run()
+	teardown()
+	os.Exit(code)
+}
+
+func setup() {
+	mockserver  = nnsmockserver.NewNnsMockServer()
+	go  mockserver.StartGrpcServer(nnsPort)
+	fmt.Printf("mock nns server started\n")
+}
+
+func teardown() {
+	mockserver.StopGrpcServer()
+	fmt.Printf("mock nns server stopped\n")
+}
+
+// CNI ADD to add container to network
+func TestAddContainerNetworking(t *testing.T) {
+
+	client := &NnsGrpcClient{}
```

Comment:
how does the client know to hit the mock server?

Reply:
Since this client is meant to be used on a host, it is written to reach out to a specific port on localhost. Test actually starts a mock nns server (which starts at the expected port), instantiates a client and invokes APIs